src/libcollections/string.rs

```diff
@@ -427,13 +463,33 @@ impl String {
     ///
     /// # Safety
     ///
-    /// This is _very_ unsafe because:
+    /// This is highly unsafe, due to the number of invariants that aren't
+    /// checked:
+    ///
+    /// * The memory at `ptr` needs to have been previously allocated by the
+    ///   same allocator the standard library uses.
+    /// * `length` needs to be less than or equal to `capacity`.
+    /// * `capacity` needs to be the correct value.
+    ///
+    /// Violating these may cause problems like corrupting the allocator's
+    /// internal datastructures.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
     ///
-    /// * We call `Vec::from_raw_parts` to get a `Vec<u8>`. Therefore, this
-    ///   function inherits all of its unsafety, see [its
-    ///   documentation](../vec/struct.Vec.html#method.from_raw_parts)
-    ///   for the invariants it expects, they also apply to this function.
-    /// * We assume that the `Vec` contains valid UTF-8.
+    /// ```
+    /// unsafe {
+    ///     let s1 = String::from("hello");
+    ///     let ptr = s1.as_ptr() as *mut _;
+    ///     let len = s1.len();
+    ///     let capacity = s1.capacity();
+    ///
+    ///     let s2 = String::from_raw_parts(ptr, len, capacity);
```

Comment:
This example isn't actually correct because it'll double-free the backing storage (e.g. ownership wasn't taken from the original string)